File: AndroidDesignTrimPath/app/src/main/java/com/example/android/androiddesigntrimpath/MainActivity.java
package com.example.android.androiddesigntrimpath;

import android.app.Activity;
import android.graphics.drawable.AnimatedVectorDrawable;
import android.os.Bundle;
import android.view.View;
import android.widget.ImageView;

public class MainActivity extends Activity {

    private ImageView imageView;
    private AnimatedVectorDrawable avd;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        imageView = (ImageView) findViewById(R.id.image_view);
        avd = (AnimatedVectorDrawable) getDrawable(R.drawable.avd_android_design);
        imageView.setImageDrawable(avd);
    }

    public void click(View view) {
        avd.start();
    }
}

File: GroupsDemo/app/src/main/java/com/udacity/groupsdemo/MainActivity.kt
package com.udacity.groupsdemo

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.support.constraint.Group
import android.view.View
import android.widget.Button

class MainActivity : AppCompatActivity(), View.OnClickListener {

    lateinit var button: Button
    lateinit var group: Group

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        button = findViewById(R.id.button)
        group = findViewById(R.id.group)

        button.setOnClickListener(this)
    }

    override fun onClick(v: View?) {
        if (group.visibility == View.GONE) {
            group.visibility = View.VISIBLE
            button.setText(R.string.hide_details)
        } else {
            group.visibility = View.GONE
            button.setText(R.string.show_details)
        }
    }
}
